core/src/main/java/io/questdb/PropServerConfiguration.java

```diff
@@ -1389,6 +1390,20 @@ public void init(CairoEngine engine, FreeOnExit freeOnExit) {
         this.factoryProvider = fpf.getInstance(this, engine, freeOnExit);
     }
 
+    private long detectRssLimitDefault() {
+        OperatingSystemMXBean mxBean = Os.getOsMXBean();
+        if (mxBean != null) {
+            long totalMemory = mxBean.getTotalPhysicalMemorySize();
+            long rssLimitDefault = totalMemory / 3 * 2;
```

Comment:
Why 2/3? On a beefy machine this means that we won't be using gigabytes of RAM. Considering that VM is lazy and overcommit is fine in many cases, we should make it something like 100% or close to it.

Reply:
The idea is to find a balance between RSS and the disk cache. But, this is open to further refinement.

Reply:
I'm still not convinced that 2/3 is a sane default. On smaller boxes where overcommit is more likely to happen, this will lead to failed operations.

Reply:
We can also introduce a lower bound. I already have similar logic when coalescing the two reported memory sizes.

Reply:
Yes, some kind of lower bound would be good to have.

Reply:
In the meantime, the curret code uses a fixed 90% of reported RAM. I fear this is too high because fragmentation could push this over 100% and result in an OOM kill.

Reply:
That's fine: 90% is better than 66.(6)%. Let's keep it 90%.